schematic/scheme/gschem/builtins.scm

```diff
@@ -310,8 +315,51 @@
 (define-action-public (&attributes-attach #:label (_ "Attach Attributes") #:icon "attribute-attach")
   (%attributes-attach))
 
-(define-action-public (&attributes-detach #:label (_ "Detach Attributes") #:icon "attribute-detach")
-  (%attributes-detach))
+
+
+( define-action-public
+  (
+    &attributes-detach
+    #:label (_ "Detach Attributes")
+    #:icon "attribute-detach"
+  )
+
+  ( let*
+    (
+    ( page ( active-page ) )
+    ( sel  ( if page (page-selection page) '() ) )
+    ( attrs '() )
+    )
+
+    ( define ( is-attr-to-detach obj ) ; predicate
```

Comment:
`is-attr-to-detach()` is a predicate, in Scheme world predicates are usually named with question mark appended.  I would recommend you to add it.  And I would suggest another name for this procedure: `detachable-attrib?`. WDYT? (Don't know if it sounds correct in English.)

Reply:
I think `detachable` is OK.